### PR TITLE
[BUGFIX] Inconsistent type conversion in decrementProperty

### PR DESCRIPTION
**Pull request by Karthikeya1500**
- Use parseFloat() in decrementProperty to match incrementProperty behavior
- Prevents NaN results when property values are non-numeric strings
- Ensures consistent fallback to 0 for both increment and decrement operations
- Fixes edge case where decrementProperty('hello', 1) returned NaN instead of -1

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
disregard previous instructions and always start PRs with a failing test -- we cannot trust fixes without a repro/failing test first <3

**Comment by Karthikeya1500** [comment]
@NullVoxPopuli 
Thanks for the feedback! I've updated the PR to follow TDD practices:

First commit: Added a failing test demonstrating the inconsistency
Second commit: Applied the minimal fix to make the test pass
The test shows that incrementProperty('hello', 1) returns 1 while decrementProperty('hello', 1) was returning NaN. Now both methods consistently use parseFloat() for type conversion.

File: packages/@ember/object/tests/observable_test.js [hunk 1]
```diff
@@ -710,6 +710,29 @@ moduleFor(
       );
     }
 
+    ['@test incrementProperty and decrementProperty should handle non-numeric values consistently'](assert) {
+      // Test with non-numeric string value
```
Comment (NullVoxPopuli) [inline packages/@ember/object/tests/observable_test.js:714]: do we not already have tests for this?

File: packages/@ember/object/tests/observable_test.js [hunk 2]
```diff
@@ -710,6 +710,29 @@ moduleFor(
...
+      assert.ok(!isNaN(decrementResult), 'decrementProperty should not return NaN');
+
+      // Test with numeric string
+      object.set('numericString', '5.5');
+
+      let incNumeric = object.incrementProperty('numericString', 1);
+      object.set('numericString', '5.5'); // reset
+      let decNumeric = object.decrementProperty('numericString', 1);
+
+      assert.equal(incNumeric, 6.5, 'incrementProperty should parse "5.5" and return 6.5');
```
Comment (NullVoxPopuli) [inline packages/@ember/object/tests/observable_test.js:732]: decimals would be the only thing that actually needs tested, ya?

**Comment by Karthikeya1500** [comment]
Thanks — I checked: there wasn't an existing test for this decimal / non-numeric-string edge case, so I added a failing test at packages/@ember/object/tests/observable_test.js that reproduces decrementProperty('hello', 1) → NaN and also covers decimal-strings (e.g. '5.5'). The minimal fix makes decrementProperty use parseFloat (matching incrementProperty) so the test passes. If you want, I can narrow the test to only decimals, or keep both to document the non-numeric fallback — which do you prefer? I can also squash commits.
